src/axis.hpp

```diff
@@ -134,25 +132,22 @@ inline void Uff(DataType* original_points,          /* in */
   if (inverse[metric_order_indices[last_index]] == -1) {
     if (!stable) {
       for (IndexingType i_dim{0}; i_dim < point_dim; i_dim++) {
-        new_points[number_of_new_points * point_dim +
-                            i_dim] =
-            original_points[metric_order_indices[last_index] *
-                                         point_dim +
-                                     i_dim];
-        new_indices[number_of_new_points] =
-            metric_order_indices[last_index];
+        new_points[number_of_new_points * point_dim + i_dim] =
+            original_points[metric_order_indices[last_index] * point_dim +
+                            i_dim];
+        new_indices[number_of_new_points] = metric_order_indices[last_index];
       }
     }
     inverse[metric_order_indices[last_index]] = number_of_new_points;
     number_of_new_points++;
     newpointmasks[metric_order_indices[last_index]] = true;
-  } else {
-    newpointmasks[metric_order_indices[last_index]] = false;
   }
 
   if (stable) {
     IndexingType counter{};
-    stable_inverse.assign(number_of_new_points, -1);
+    // This could be a map (but I think this is sufficient for the moment,
+    // especially if only a few number of duplicates exist)
+    stable_inverse.assign(number_of_points, -1);
```

Comment:
doch n_points?

Reply:
Actuwlly would be the highest non-duplicate Index, but I didn't want to keep track of yet another value. However, we could use a `std::map` here to save some memory

Reply:
Actuwlly I am ok with this